raiden/utils/__init__.py

```diff
@@ -288,3 +289,36 @@ def is_frozen():
 
 def event_decoder(event: Log, contract_translator: ContractTranslator):
     return contract_translator.decode_event(event.topics, event.data)
+
+
+ResultCache = namedtuple('ResultCache', ('result', 'timestamp'))
+
+
+def cache_response_timewise(seconds=600):
+    """ This is a decorator for caching results of functions of objects.
+    The objects must have:
+        - A results_cache dictionary attribute
+        - A semaphore attribute named lock
+
+    Objects adhering to this interface are all the exchanges and the rotkelchen object.
```

Comment:
That comment should be updated.

Reply:
ops. hahaah I just noticed. Copied this from another project of mine :)